index.js

```diff
@@ -24,64 +26,93 @@ function getRandomNumber(min, max) {
 
 function checkGuess() {
   // Get value from guess input element
-  const guess = parseInt(guessInput.value, 10);
-  attempts = attempts + 1;
-
+  let guess = parseInt(guessInput.value, 10);
+  guessInput.value='';
+  
   hideAllMessages();
-
+  resetButton.style.display = ''
+  
   if (guess === targetNumber) {
-    numberOfGuessesMessage.style.display = '';
     numberOfGuessesMessage.innerHTML = `You made ${attempts} guesses`;
-
+    numberOfGuessesMessage.style.display='block';
     correctMessage.style.display = '';
-
     submitButton.disabled = true;
+    
     guessInput.disabled = true;
+    
+    resetButton.style.display = '';
+    
   }
+  
+   
 
-  if (guess !== targetNumber) {
-    if (guess < targetNumber) {
-      tooLowMessage.style.display = '';
-    } else {
-      tooLowMessage.style.display = '';
-    }
+  if (guess < targetNumber) {
+    tooLowMessage.style.display = '';
+    
+  }
+  
+   
+   
+  if (guess > targetNumber) {
+    tooHighMessage.style.display = '';
+    
+  }
 
-    const remainingAttempts = maxNumberOfAttempts - attempts;
 
-    numberOfGuessesMessage.style.display = '';
-    numberOfGuessesMessage.innerHTML = `You guessed ${guess}. <br> ${remainingAttempts} guesses remaining`;
-  }
+  const remainingAttempts = maxNumberOfAttempts - attempts;
+  console.log(remainingAttempts);
 
-  if (attempts ==== maxNumberOfAttempts) {
-    submitButton.disabled = true;
-    guessInput.disabled = true;
+  if(guess !==targetNumber){
+    
+
+    maxGuessesMessage.innerHTML = `You guessed ${guess} <br>   ${remainingAttempts} guesses remaining`;
+  maxGuessesMessage.style.display = '';
+  
+    
+}
+
+else{
+  document.getElementById('max-guesses');
+  
+  //submitButton.disabled=true;
+   //guessInput.disabled=true;
+ }
+ if(attempts===maxNumberOfAttempts){
+  guessInput.disabled=true;
+  submitButton.disabled=true;
   }
+    attempts++;
+  }
+  
+
 
-  guessInput.value = '';
 
-  resetButton.style.display = '';
-}
 
 function hideAllMessages() {
-  for (let elementIndex = 0; elementIndex <= messages.length; elementIndex++) {
+  for (let elementIndex = 0; elementIndex < messages.length; elementIndex++) {
     messages[elementIndex].style.display = 'none';
+
   }
+
+
 }
 
-funtion setup() {
+function setup() {
   // Get random number
   targetNumber = getRandomNumber(1, 100);
   console.log(`target number: ${targetNumber}`);
 
   // Reset number of attempts
```

Comment:
Right now, you're not actually setting the number of `attempts`. So if I play one time and it takes me 3 guesses to get the right answer, and then I click "Reset" button, then I try to play again but it immediately already thinks I only have `2 guesses remaining` which shouldn't be the case. The code should make sure to reset `attempts` back to `0` when someone clicks reset.

Reply:
i added attempts=0; to reset number of attempts